base/mapiterator.jl

```diff
@@ -0,0 +1,175 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+"""
+    MapIterator(f, c...) -> iterator
+
+`collect(MapIterator(f, c...))` is equivalent to `map(f, c...)`
+"""
+
```

Comment:
Looks like it's not required to get docstrings to be associated with the following function, but stylistically it makes the association less obvious.